be/src/storage/lake/persistent_index_memtable.cpp

```diff
@@ -51,15 +52,15 @@ Status PersistentIndexMemtable::upsert(size_t n, const Slice* keys, const IndexV
 }
 
 Status PersistentIndexMemtable::insert(size_t n, const Slice* keys, const IndexValue* values, int64_t version) {
-    TRACE_COUNTER_SCOPE_LATENCY_US("persistent_index_memtable_insert_us");
+    TRACE_COUNTER_SCOPE_LATENCY_US("pindex_mem_insert_us");
     for (size_t i = 0; i < n; ++i) {
-        auto key = keys[i].to_string();
+        auto key = std::string_view(keys[i]);
         auto size = keys[i].get_size();
         const auto value = values[i];
         std::list<IndexValueWithVer> index_value_vers;
         index_value_vers.emplace_front(version, value);
         if (auto [it, inserted] = _map.emplace(key, index_value_vers); inserted) {
-            _keys_size += key.capacity() + sizeof(std::string);
+            _keys_size += key.size() + sizeof(std::string);
```

Comment:
`size()` is smaller than `capacity()`, if you change from capacity to size, it will be smaller than current way. Is it what you want?

Reply:
This key is a slice not strig.

Reply:
But when you put this into the map, A string will be constructed.

Reply:
I will check the code, but I think `phmap` has no reason to reserve more memory than it needs.